common/src/main/java/com/wynntils/features/inventory/CustomBankPagesFeature.java

```diff
@@ -219,19 +244,61 @@ private void clickPreviousPage() {
 
     @Override
     protected void onConfigUpdate(ConfigHolder configHolder) {
-        switch (configHolder.getFieldName()) {
-            case "buttonOnePage",
-                    "buttonFourPage",
-                    "buttonFivePage",
-                    "buttonSixPage",
-                    "buttonThreePage",
-                    "buttonTwoPage" -> {
-                if ((int) configHolder.getValue() < 1) {
-                    configHolder.setValue(1);
-                } else if ((int) configHolder.getValue() > MAX_BANK_PAGES) {
-                    configHolder.setValue(MAX_BANK_PAGES);
-                }
+        String valueString = (String) configHolder.getValue();
+
+        List<Integer> originalValues = parseStringToDestinations(valueString);
+
+        if (originalValues == null) {
+            configHolder.setValue(configHolder.getDefaultValue());
+            return;
+        }
+
+        int maxValue =
+                configHolder.getFieldName().equals("bankDestinations") ? MAX_BANK_PAGES : MAX_HOUSING_CONTAINER_PAGES;
+
+        List<Integer> newValues = modifyJumpValues(originalValues, maxValue);
+
+        if (!newValues.equals(originalValues)) {
+            String formattedConfig = newValues.stream().map(Object::toString).collect(Collectors.joining(","));
+
+            configHolder.setValue(formattedConfig);
+        }
+    }
+
+    private List<Integer> parseStringToDestinations(String destinationsStr) {
+        String[] destinationStrings = destinationsStr.split(",");
+
+        if (destinationStrings.length != 6) {
+            return null;
+        }
+
+        List<Integer> destinations = new ArrayList<>();
+
+        for (String destinationString : destinationStrings) {
```

Comment:
I think this can be written in a more modern Java style using functional expressions and stream. Something along the lines of `destinationStrings.stream.map(Integer::parseInt).toList()`.